smartsim/settings/batchSettings.py

```diff
@@ -0,0 +1,240 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2024, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import annotations
+import typing as t
+import copy
+
+from smartsim.log import get_logger
+from .._core.utils.helpers import fmt_dict
+from .common import validate_env_vars, validate_args, StringArgument
+from .batchCommand import SchedulerType
+from .translators.batch.pbs import QsubBatchArgTranslator
+from .translators.batch.slurm import SlurmBatchArgTranslator
+from .translators.batch.lsf import BsubBatchArgTranslator
+from .translators import BatchArgTranslator
+from .baseSettings import BaseSettings
+
+logger = get_logger(__name__)
+
+class BatchSettings(BaseSettings):
+    def __init__(
+        self,
+        batch_scheduler: t.Union[SchedulerType, str],
+        scheduler_args: t.Optional[t.Dict[str, t.Union[str,int,float,None]]] = None,
+        env_vars: t.Optional[StringArgument] = None,
+    ) -> None:
+        try:
+            self._batch_scheduler = SchedulerType(batch_scheduler)
+        except KeyError:
+            raise ValueError(f"Invalid scheduler type: {batch_scheduler}")
+        self._arg_translator = self._get_scheduler()
+
+        if env_vars:
+            validate_env_vars(env_vars)
+        self.env_vars = env_vars or {}
+
+        if scheduler_args:
+            validate_args(scheduler_args)
+        self.scheduler_args = scheduler_args or {}
```

Comment:
Per offline discussion: it was decided to remove this attr and have the translators manage their own internal representation of partially built batch args